airbyte-integrations/connectors/source-mssql/src/testFixtures/java/io/airbyte/integrations/source/mssql/MsSQLTestDatabase.java

```diff
@@ -68,12 +71,42 @@ static public MsSQLTestDatabase in(final BaseImage imageName, final ContainerMod
 
   public MsSQLTestDatabase(final MSSQLServerContainer<?> container) {
     super(container);
+    LOGGER.info("creating new database. databaseId=" + this.databaseId + ", databaseName=" + getDatabaseName());
   }
 
   public MsSQLTestDatabase withCdc() {
     return with("EXEC sys.sp_cdc_enable_db;");
   }
 
+  public synchronized MsSQLTestDatabase withCdcForTable(String schemaName, String tableName, String roleName) {
+    String captureInstanceName = "%s_%s".formatted(schemaName, tableName);
+    return withCdcForTable(schemaName, tableName, roleName, captureInstanceName);
+  }
+
+  public synchronized MsSQLTestDatabase withCdcForTable(String schemaName, String tableName, String roleName, String captureInstanceName) {
+    final var enableCdcSqlFmt = """
+                                EXEC sys.sp_cdc_enable_table
+                                \t@source_schema = N'%s',
+                                \t@source_name   = N'%s',
+                                \t@role_name     = %s,
+                                \t@supports_net_changes = 0,
+                                \t@capture_instance = N'%s'""";
+    String sqlRoleName = roleName == null ? "NULL" : "N'%s'".formatted(roleName);
+
+    for (int i = 0; i < MAX_RETRIES; i++) {
+      try {
+        getDslContext().execute(enableCdcSqlFmt.formatted(schemaName, tableName, sqlRoleName, captureInstanceName));
+        return this;
+      } catch (Exception e) {
+        if (!e.getMessage().contains("The error returned was 14258: 'Cannot perform this operation while SQLServerAgent is starting.")) {
+          throw new RuntimeException(e);
+        }
+        Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+      }
+    }
+    throw new RuntimeException("Couldn't enable CDC for table %s.%s".formatted(schemaName, tableName));
+  }
```

Comment:
Nice.
Do we want to only do this waiting at most once per container, perhaps? Any time after the first call the stored procedure should just work on the first try.

Reply:
good question. As far as I can tell, once the procedure worked for a DB, it'll work for the next one. One caveat is that the same procedure can't be run twice at the same time for the same DB, hence the synchronized keyword

Reply:
I'm fine with just leaving it up to a global timeout to handle this edge case. I noticed the `synchronized`, nicely done!

Reply:
I did some more testing, and I don't think it's always working after the 1st table of a container has been initialized. I really think it's at least once per DB...